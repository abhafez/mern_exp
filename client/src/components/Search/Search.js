import React from 'react';
import { Button } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Input = styled.input`
  display: inline;
  margin-right: 1em;
`;

function Search({ value, getName, searchByName, loading }) {
  return (
    <>
      <Input
        onChange={getName}
        value={value}
        id="name-search"
        placeholder="Product Name"
      />
      <Button
        id="search-btn"
        type="submit"
        data-testid="btn"
        onClick={searchByName}
        loading={loading}
      >
        Search by Name
      </Button>
    </>
  );
}

Search.propTypes = {
  value: PropTypes.string,
  getName: PropTypes.func,
  searchByName: PropTypes.func,
  loading: PropTypes.bool,
};
export default Search;
