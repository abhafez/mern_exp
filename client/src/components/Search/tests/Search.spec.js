import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Search from '../Search';

test('Loads when clicked', () => {
  const searchSpy = jest.fn();
  const getNameSpy = jest.fn();
  const { getByText, container } = render(
    <Search
      value="name"
      loading={!!true}
      searchByName={searchSpy}
      getName={getNameSpy}
    />,
  );
  expect(getByText('Search by Name')).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
  // Passing loading={true}
  expect(screen.getByTestId('btn').classList).toContain('loading');

  // Clicking search button
  expect(searchSpy).toBeCalledTimes(0);
  fireEvent.click(getByText(/Search by Name/));
  expect(searchSpy).toBeCalledTimes(1);
});

test('No loading by default', () => {
  const searchSpy = jest.fn();
  const getNameSpy = jest.fn();

  render(
    <Search value="name" loading={false} searchByName={searchSpy} getName={getNameSpy} />,
  );
  expect(screen.getByTestId('btn').classList).not.toContain('loading');
});
