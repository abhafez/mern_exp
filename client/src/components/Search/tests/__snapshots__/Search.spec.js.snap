// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads when clicked 1`] = `
<input
  class="sc-AxjAm iMmbHG"
  id="name-search"
  placeholder="Product Name"
  value="name"
/>
`;
